modin/backends/pandas/query_compiler.py

```diff
@@ -562,12 +562,13 @@ def is_monotonic_decreasing(self):
         lambda df, **kwargs: df.apply(
             lambda x: (x.sum(skipna=kwargs.get("skipna", True)), x.count()),
             axis=kwargs.get("axis", 0),
-        ),
+            result_type="reduce",
+        ).set_axis(df.axes[kwargs.get("axis", 0) ^ 1], axis=0),
```

Comment:
The `apply` functionality was changed to automatically try to coerce lists to `DataFrames`. When that happens the index returns to the default `RangeIndex`. This sets the `index` back to the correct index based on `axis` argument, because we are always keeping the opposite axis, which is why the XOR (`axis ^ 1`).